build/stage-build-samples.yml

```diff
@@ -0,0 +1,124 @@
+jobs:
+- job: VS_Latest
+  timeoutInMinutes: 600
+
+  pool:
+    vmImage: windows-2022
+
+  steps:
+  - checkout: self
+    clean: true
+  
+  - powershell: |
+      if($env:System_PullRequest_TargetBranch -eq $null) { $TargetBranch = "master" } else { $TargetBranch = $env:System_PullRequest_TargetBranch; }
+
+      $dict = New-Object 'System.Collections.Generic.Dictionary[String,System.Collections.Generic.Dictionary[String,String]]'
+      $samples = Get-ChildItem -Path **\*.sln -Recurse | Where-Object {$_.FullName -notmatch "\\ArchivedProjects\\"}
+      foreach($sample in $samples){
+
+        $solutionPath=[System.IO.Path]::GetDirectoryName($sample.FullName);
+        Write-Host "$solutionPath"
+
+        git diff --quiet HEAD "origin/$TargetBranch" -- "$solutionPath"
+
+        # Only build if there a change in the solution path for the current PR, if we're not in a PR
+        if( ("$env:System_PullRequest_PullRequestId" -eq '') -or ($global:LASTEXITCODE -ne 0)) {
+          Write-Host "Adding $solutionPath"
+          $item = New-Object 'System.Collections.Generic.Dictionary[String,String]'
+          $item.Add("solutionPath", $sample.FullName)
+          $name = $sample.Name.Split(".")[0]
+          if(!$dict.ContainsKey($name)){
+            $dict.Add($name, $item)
+          }
+        }
+        $global:LASTEXITCODE = 0
+
+      }
+
+      $solutionsJson = $dict | ConvertTo-Json -Compress
+      
+      Write-Host $solutionsJson
+      Write-Host "##vso[task.setvariable variable=samplesJson;isOutput=true]$solutionsJson"
+
+      Write-Host "##vso[task.setvariable variable=solutionPath;isOutput=true]$solutionPath"
+    name: generateJson
+    displayName: Generate Json of Samples
+
+- job: Build
+  dependsOn: VS_Latest
+
+  pool:
+    vmImage: 'windows-2022'
+
+  strategy:
+    # matrix:
+    #   Commerce:
+    #     solutionPath: 'D:\a\1\s\reference\Commerce\Commerce.sln'
+    #   NotChanged:
+    #     solutionPath: 'D:\a\1\s\UI\AdvancedXBind\AdvancedXBind.sln'
+    matrix: $[ dependencies.VS_Latest.outputs['generateJson.samplesJson'] ]
+    
+  variables:
+    - name: UseDotNetNativeToolchain
+      value: false
+    - name: solutionPath
+      value: $[ dependencies.VS_Latest.outputs['generateJson.solutionPath'] ] 
+
+  steps:
+  - checkout: self
+    clean: 'true'
+
+  - template: templates/dotnet-install-windows.yml
+
+  - ${{ if startsWith(variables['Build.SourceBranch'], 'refs/heads/canaries') }}:
+    - template: .vsts-ci.canaries.yml
+      parameters:
+        solution: $(solutionPath)
+
+  - powershell: |
+        & choco install vswhere
+        $msbuildpath=(vswhere -latest -requires Microsoft.Component.MSBuild -find MSBuild\**\Bin\MSBuild.exe -nologo)
+
+        echo "##vso[task.setvariable variable=msbuildpath]$msbuildpath"
+    displayName: Set MSBUILDPATH
```

Comment:
Not sure if this step is really needed. We are using `dotnet` directly so I'm guessing it shouldn't be needed?

Reply:
You are right @Youssef1313, I reviewed the logic and it is not needed + the variable was not used.